src/utils.php

```diff
@@ -35,3 +35,34 @@ function create_list_of_cars(array $cars): string | false
 
     return $ret;
 }
+
+function validateDataFields(Car $car): bool
+{
+    $ret = true;
+
+    if (empty($car->name) == true || strlen($car->name) > 30 || strlen($car->name) <= 0) {
+        echo "Name cannot be over 30 characters long" . "<br>";
+        $ret = false;
+    }
+
+    if (empty($car->year_made) == true || is_int($car->year_made) == false || strlen(strval($car->year_made)) <= 0 || strlen(strval($car->year_made)) > 4 || $car->year_made < 0 || $car->year_made > 2023) {
+        echo "Invalid year" . "<br>";
+        $ret = false;
+    }
+
+    if (empty($car->zero_sixty) == true || is_int($car->zero_sixty) == false || $car->zero_sixty <= 0) {
+        echo "Invalid zero to sixty value" . "<br>";
+        $ret = false;
+    }
+
+    if (empty($car->price) == true || is_int($car->price) == false || $car->price <= 0) {
+        echo "Invalid price" . "<br>";
+        $ret = false;
+    }
+
+    if (empty($car->brand) == true || strlen($car->brand) >= 20 || strlen($car->brand) <= 0) {
+        echo "Brand name cannot be over 20 characters" . "<br>";
+        $ret = false;
+    }
+    return $ret;
+}
```

Comment:
You should make this function return any error messages instead of echoing them - remember it's generally bad practice to echo from a function or method. 

Because this function echos it is not currently testable, and we ultimately want as much of our code to be testable (and tested!) as possible